metals/src/main/scala/scala/meta/internal/parsing/BloopDiagnosticsParser.scala

```diff
@@ -0,0 +1,73 @@
+package scala.meta.internal.parsing
+
+import java.nio.file.Paths
+
+import scala.jdk.CollectionConverters.SeqHasAsJava
+
+import org.eclipse.lsp4j.Diagnostic
+import org.eclipse.lsp4j.DiagnosticSeverity
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.PublishDiagnosticsParams
+import org.eclipse.lsp4j.Range
+
+case class FileDiagnostic(filepath: String, diagnostic: Diagnostic)
+
+object BloopDiagnosticsParser {
```

Comment:
Would probably be worth having some specific unit tests for particular different outputs